### PR TITLE
Normalize filenames

### PR DESCRIPTION
**Pull request by elahti**

### CODE REVIEW

File: src/liiteri/db/file_metadata_store.clj [hunk 1]
```diff
@@ -19,11 +19,17 @@
                              {:connection conn-arg#})]
      ~@body))
 
+(def ^:private non-valid-character-pattern #"(?i)[^a-z\.\-_0-9 ]")
+
+(defn- normalize [string]
+  (clojure.string/replace string non-valid-character-pattern ""))
```
Comment (wolverian) [inline src/liiteri/db/file_metadata_store.clj]: Does this replace _every_ occurrence of evil characters in the string, or just the first?

File: src/liiteri/db/file_metadata_store.clj [hunk 2]
```diff
@@ -46,7 +53,8 @@
   {:pre [(map? conn)
          (contains? conn :connection)]} ; force transaction
   (->> (sql-get-unscanned-file {} conn)
-       (map db-utils/unwrap-data)
+       (eduction (map db-utils/unwrap-data)
```
Comment (wolverian) [inline src/liiteri/db/file_metadata_store.clj]: What's going on here? 😕

File: test/liiteri/file_metadata_store_test.clj [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  (:require [clojure.test :refer :all]
+            [liiteri.db.file-metadata-store :as metadata-store]))
+
+(deftest normalize-filename
+  (doseq [[filename expected] [["parrot.png" "parrot.png"]
+                               ["parrot..png" "parrot..png"]
+                               ["parrot/:;.png" "parrot.png"]
+                               ["p a r r o t . p n g" "p a r r o t . p n g"]
+                               ["paRRot1234.png" "paRRot1234.png"]
+                               ["pa:-_rrot.png" "pa-_rrot.png"]]]
```
Comment (wolverian) [inline test/liiteri/file_metadata_store_test.clj]: I'd add a few Unicode tests here too, for characters that look like regular ascii but are not really.

File: src/liiteri/db/file_metadata_store.clj (hunk 1 above)
Comment (elahti) [inline src/liiteri/db/file_metadata_store.clj]: Yes

File: src/liiteri/db/file_metadata_store.clj (hunk 2 above)
Comment (elahti) [inline src/liiteri/db/file_metadata_store.clj]: Exlained irl

**Pull request merged**